lib/service_variables.rb

```diff
@@ -77,6 +82,10 @@ def string_option(name, default: nil, enum: nil)
 
   def get(key)
     redis.hget(redis_hash_key, key)
+  rescue Redis::CannotConnectError => e
+    raise e if @failure_mode == :raise_exception
+    return nil if @failure_mode == :use_default
```

Comment:
can you put a comment here saying that if Redis returns `nil` then the default value is used.  Something to just explain why `nil` works.

Reply:
yup